src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -352,6 +355,17 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 			_ => UnhandledSyntax( des )
 		};
 
+	private SeparatedSyntaxList<CollectionElementSyntax> Transform(
+		SeparatedSyntaxList<CollectionElementSyntax> original
+	) => SyntaxFactory.SeparatedList(
+		original.Select( elem => elem switch {
+			ExpressionElementSyntax ee => ee.WithExpression( Transform( ee.Expression ) ),
+			SpreadElementSyntax sp => sp.WithExpression( Transform( sp.Expression ) ),
+			_ => UnhandledSyntax( elem )
+		} ),
+		original.GetSeparators()

```

Comment:
We're not passing the original separators for the other cases in the transformer, so perhaps something to correct later

Reply:
Yeah -- see note below.

One other reason we're not doing it in the other ones is TransformAll supports the "maybe transform" e.g. so we can delete cancellation token arguments. There needs to be some accounting for that otherwise .GetSeparators() will possibly insert extra separators/commas.